command/agent/check_test.go

```diff
@@ -65,19 +71,38 @@ func expectStatus(t *testing.T, script, status string) {
 }
 
 func TestCheckMonitor_Passing(t *testing.T) {
-	expectStatus(t, "exit 0", structs.HealthPassing)
+	expectStatus(t, "exit 0", "", structs.HealthPassing)
+}
+
+func TestCheckMonitor_Chroot(t *testing.T) {
+	// running this test only on linux and if the uid is root
+	if runtime.GOOS != "linux" || syscall.Getuid() != 0 {
+		return
+	}
+	// Creating the chroot
+	chroot, err := ioutil.TempDir("", "consulchroot")
+	defer os.RemoveAll(chroot)
```

Comment:
This could probably be moved to after the error check, assuming that error means it couldn't be created.